### PR TITLE
ING-523: Improved GetAllReplicas handling

### PR DESCRIPTION
**Pull request by Westwooo**
Although the GetAllReplicas change in gocbcorex means that individual replica reads can push an error to the channel this hasn't been propagated to STG, since doing so would cause a change in behaviour between traditional GetAllReplicas 
and GetAllReplicas through protostellar.  I think this would be a nice change in future, but if we do so we should update all GetAllReplicas, so for the time being STG will behave the same as core GetAllReplicas and just log any individual replica read errors.

### CODE REVIEW

File: gateway/dataimpl/server_v1/kvserver.go [hunk 1]
```diff
@@ -1394,156 +1394,43 @@ func (s *KvServer) GetAllReplicas(in *kv_v1.GetAllReplicasRequest, out kv_v1.KvS
...
-		remainingReads--
-
-		if nextRes.res != nil {
-			err := out.Send(nextRes.res)
-			if err != nil {
-				asyncReadRemaining()
-				return s.errorHandler.NewGenericStatus(err).Err()
-			}
-		}
+	if err != nil {
```
Comment (chvck) [inline gateway/dataimpl/server_v1/kvserver.go:1430]: I think this is a change in error handling behaviour? Previously if the first read failed then we'd return an error. I also think we should just put the error checking blocks next to the error assignment. It's a bit more verbose but much easier to read/understand intent.

Comment (brett19) [reply gateway/dataimpl/server_v1/kvserver.go:1430]: Doesn't this still return an error on the first read failure?  Since the for loop above is dependent on `res=nil`?

Comment (chvck) [reply gateway/dataimpl/server_v1/kvserver.go:1430]: Ah I misunderstood this. I do find it a bit hard to follow and think we should just use a for {} loop without calling .Next() before it. Then we can just handle the errors next to the error being returned.

Comment (brett19) [reply gateway/dataimpl/server_v1/kvserver.go:1430]: Filed https://couchbasecloud.atlassian.net/browse/ING-804 for future improvements to this.
